### ISSUE TITLE
How to select nodes for collections 

### ISSUE THREAD

**Issue by yjl9122** [openai/gradient-checkpointing#26]
I add all tf.nn.relu to ‘checkpoints', and I also have weights_regularizer. I don’t save memory, so Is there something wrong?

**Comment by yjl9122** [comment]
@yaroslavvb would you give me some suggestions?
